jupyter/scripts/incremental_learning/generate_experiment_matrix.py

```diff
@@ -0,0 +1,117 @@
+import argparse
+import copy
+import csv
+import json
+import random
+from incremental_learning.config import datasets_dir
+from incremental_learning.config import configs_dir
+from incremental_learning.storage import download_dataset
+
+def feature_fields(dataset_name : str):
+    with open(datasets_dir / '{}.csv'.format(dataset_name)) as csv_file:
+        reader = csv.reader(csv_file, delimiter=',')
+        field_names = next(reader)
+    with open(configs_dir / '{}.json'.format(dataset_name)) as json_file:
+        config = json.load(json_file)
+    target = [config['analysis']['parameters']['dependent_variable']]
+    categorical_features = []
+    if 'categorical_fields' in config:
+        categorical_features = [name for name in config['categorical_fields'] if name not in target]
+    metric_features = [name for name in field_names if name not in categorical_features + target]
+    return target, metric_features, categorical_features
+
+def features():
+    result = {}
+    for dataset_name in args.classification_datasets:
+        if download_dataset(dataset_name):
+            result[dataset_name] = feature_fields(dataset_name)
+    for dataset_name in args.regression_datasets:
+        if download_dataset(dataset_name):
+            result[dataset_name] = feature_fields(dataset_name)
+    return result
+
+def needs_metric_features(transform_name: str):
```

Comment:
Not sure what is going on in this method - are you checking that the supplied `transform_name` is one of the approved metric transforms?

Reply:
This lists out all transforms which absolutely require that there is at least one metric feature (we don't generate the experiment against datasets which only have categorical features). I added documentation in [72c2a29](https://github.com/elastic/ml-cpp/pull/1994/commits/72c2a290fe948e6e8341da565c561adae57ffbb2).